### PR TITLE
add secp-192k1

### PR DESCRIPTION
**Pull request by xsilen**
I don't know much about ecc, please review my pull reqest or notice me anything what i missed

### CODE REVIEW

**Comment by xsilen** [comment]
test failed.
the 192k1 works fine in my project, so i want to integrate into phpecc
i don't know what's the relationship between Secp256r1 with NistP256, i just copy `testSecp256r1EquivalenceToNistP256` to `testSecp192k1EquivalenceToNistP192`.

```
 public function testSecp192k1EquivalenceToNistP192(GmpMathInterface $adapter)
    {
        $secpFactory = EccFactory::getSecgCurves($adapter);
        $nistFactory = EccFactory::getNistCurves($adapter);

        $signer = new Signer($adapter);

        $secret = gmp_init('DC51D3866A15BACDE33D96F992FCA99DA7E6EF0934E7097559C27F1614C88A7F', 16);

        $secpKey = $secpFactory->generator192k1()->getPrivateKeyFrom($secret);
        $nistKey = $nistFactory->generator192()->getPrivateKeyFrom($secret);

        $randomK = RandomGeneratorFactory::getRandomGenerator()->generate($secpKey->getPoint()->getOrder());
        $message = RandomGeneratorFactory::getRandomGenerator()->generate($secpKey->getPoint()->getOrder());

        $sigSecp = $signer->sign($secpKey, $message, $randomK);
        $sigNist = $signer->sign($nistKey, $message, $randomK);

        $this->assertTrue($adapter->equals($sigNist->getR(), $sigSecp->getR()));
        $this->assertTrue($adapter->equals($sigNist->getS(), $sigSecp->getS()));
    }
```

File: tests/specs/secp-192k1.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    x: 6B2C6D6A14D4230886210E73316660D3C1941553DE3DC5B9
+    y: BB29C72DCA52F16A9BCDED0CEA61FA5A10F178CD4F32F65C
+
+diffie:
+  -
+    alice: 1322467643339359286869087270951059000985239245302514649815
+    bob: 1939305438156392176366286612110560319900430562044791698413
+    shared: f0dce7a57d35b8518c106221adc8987fbb46d29c840febcb
+
+hmac:
```
Comment (afk11) [inline tests/specs/secp-192k1.yml]: These look like the test fixtures for secp256k1, except k, r, s are shorter. Did you make these yourself or can you link where you got them?

**Comment by afk11** [comment]
Can you link where you got the curve parameters and I'll look into the equivalence tests?

**Comment by afk11** [comment]
There's definitely a higher burden when submitting a new curve, please give me all the information you can - where you got the parameters, and where you found the test vectors. 

If accepting a new curve to the library I need to know whether the test vectors are taken from somewhere official, or just generated using the library itself. From the HMAC stuff, it looks like you generated them yourself, which isn't helpful because we already know the library works, we need to know if it has any edge cases that cause it to deviate from the official set. 

Same for keypairs / ecdh.

**Comment by xsilen** [comment]
i just craft a convert script 
```
$source = <<<EOL
static const mbedtls_mpi_uint secp192k1_p[] = {
    BYTES_TO_T_UINT_8( 0x37, 0xEE, 0xFF, 0xFF, 0xFE, 0xFF, 0xFF, 0xFF ),
    BYTES_TO_T_UINT_8( 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF ),
    BYTES_TO_T_UINT_8( 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF ),
};
EOL;

preg_match_all('/(?<=0x)[0-9a-fA-F]{2}/', $source, $matches);

$source = implode('', $matches[0]);
var_dump($source);

$source = bin2hex(strrev(hex2bin($source)));
var_dump($source);

$result = gmp_strval(gmp_init($source, 16), 10);

echo $result;
```

CurveParams  can be found here 
https://github.com/ARMmbed/mbedtls/blob/development/library/ecp_curves.c#L294-L298
params convert result: `6277101735386680763835789423207666416102355444459739541047`
gx convert result: `db4ff10ec057e9ae26b07d0280b7f4341da5d1b1eae06c7d`
gy convert result: `9b2f2f6d9c5628a7844163d015be86344082aa88d95e2f9d`
n convert result:   `8DFDDE746A46690F17FCF226FEFFFFFFFFFFFFFFFFFFFFFF`

and i use online ecdh test 
http://www-cs-students.stanford.edu/~tjw/jsbn/ecdh.html

**Comment by afk11** [comment]
Noting steps as I do them, we've never had a request for a new curve in a long time!

- [x] Verify CurveFp and GeneratorPoint parameters (SecGcurve) (http://www.secg.org/sec2-v2.pdf)
- [ ] Verify qualifications for NamedCurveFp 
   - [x] Check name, and OID (http://www.secg.org/sec2-v2.pdf)
   - [ ] Check Openssl uses same name
- [ ] Test vectors: 
   - [ ] Key generation (reputable fixtures required)
   - [ ] EcDH (less important) 
   - [ ] ECDSA (same as key generation)
   - [ ] HMAC-DRBG (we've accepted unofficial ones before, for widely accepted ones..)

File: tests/specs/secp-192k1.yml (hunk 1 above)
Comment (xsilen) [inline tests/specs/secp-192k1.yml]: yes, i don't know about hmac exactly. i just copy from secp256k1.

**Comment by afk11** [comment]
I think we can accept it with reputable generation vectors, ECDSA and EcDH will probably do with vectors from other implementations, and there isn't a hope of finding HMACDRBG ones, so don't worry about them..

**Comment by xsilen** [comment]
so i have removed hmac from test case. and add a test case from  http://cryptomanager.com/tv.html

**Comment by afk11** [comment]
Added ECDSA fixture from http://cryptomanager.com/tv.html

**Comment by afk11** [comment]
Only missing HMACDRBG fixtures, but we can live without those. Added commits to use PHP7 return types to the new feature. Backported this PR to v0.4. https://github.com/phpecc/phpecc/pull/193

**Pull request merged**